File: C++/level2/n진수_게임.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

string solution(int n, int t, int m, int p) {
    string answer = "";
    string order = "0123456789ABCDEF";
    int val = 0;
    string result = "0";

    while(result.size() <= t*m){
        string temp = "";
        int valTemp = val;
        while(val != 0){
            temp = order[val%n] + temp;
            val /= n;
        }

        result += temp;
        val = valTemp + 1;
    }

    for(int i=0; i<t; i++)
        answer += result[m*i+p-1];

    return answer;
}

File: JavaScript/level2/n진수_게임.js
function solution(n, t, m, p) {
    var answer = '';
    let stringVal = "";
    let val = 0;
    
    while(stringVal.length < t*m){
        stringVal += val.toString(n);
        val++;
    }
    
    stringVal = stringVal.toUpperCase()
    
    for(let i=0; i<t; i++)
       answer += stringVal[i*m+p-1];
    
    return answer;
}
